commcare_connect/opportunity/views.py

```diff
@@ -110,6 +110,16 @@ def test_func(self):
         ) or self.request.user.is_superuser
 
 
+def get_opportunity_or_404(pk, org_slug):
+    opp = get_object_or_404(Opportunity, id=pk)
+    if opp.organization is not None and opp.organization.slug == org_slug:
+        return opp
+    elif opp.managed and opp.managedopportunity.program.organization.slug == org_slug:
+        return opp.managedopportunity
+    else:
+        raise Http404("Opportunity not found.")
+
+
```

Comment:
This is done to ensure that the same creation flow can be used for both opportunities and managed opportunities. Since a managed opportunity will not have an organization initially, we check the program's organization in the case of a managed opportunity to avoid 404 error.

we don't need to return `opp.managedopportunity` so changed the code little bit here [276fef0](https://github.com/dimagi/commcare-connect/pull/379/commits/276fef074507256f35a10eea4ce4f948cf5be67b)